File: cmstestsuite/tasks/communication_many_stdio_stubbed/code/stub.cpp
#include <assert.h>
#include <stdio.h>

extern int userfunc1(int x);
extern int userfunc2(int x);

int main(int argc, char **argv) {
    while (1) {
        int n;
        int ret = scanf("%d", &n);
        assert(ret == 1);
        if (n == 0)
            break;
        if(argv[1][0] == '0')
            printf("correct %d\n", userfunc1(n));
        else
            printf("correct %d\n", userfunc2(n));
        fflush(stdout);
    }
    return 0;
}

File: cmstestsuite/tasks/communication_many_stdio_stubbed/code/stub.py
import sys

import user1
import user2


while True:
    n = sys.stdin.readline().strip()
    if n == "":
        # EOF, let's quit.
        break

    n = int(n)
    if n == 0:
        break
    if int(sys.argv[1]) == 0:
        sys.stdout.write("correct %d\n" % user1.userfunc1(n))
    else:
        sys.stdout.write("correct %d\n" % user2.userfunc2(n))
    sys.stdout.flush()
